core/Pos/Binary/Class/Core.hs

```diff
@@ -68,6 +70,16 @@ matchSize requestedSize lbl actualSize =
   when (actualSize /= requestedSize) $
     fail (lbl <> " failed the size check. Expected " <> show requestedSize <> ", found " <> show actualSize)
 
+-- | Tries to decode Tag 24 (CBOR Data Item) from a ByteString blob,
```

Comment:
That's right, the proper use of the tag is in front of (consider to be enclosing if you think of a term tree) the the bytestring. So not inside.